sound/soc/sof/core.c

```diff
@@ -323,28 +347,31 @@ static int sof_probe(struct platform_device *pdev)
 			 "warning: failed to initialize trace %d\n", ret);
 	}
 
+	/* vFE doesn't support pm currently */
 	/* autosuspend sof device */
 	pm_runtime_mark_last_busy(sdev->dev);
 	pm_runtime_put_autosuspend(sdev->dev);
 
 	/* autosuspend pci/acpi/spi device */
 	pm_runtime_mark_last_busy(plat_data->dev);
 	pm_runtime_put_autosuspend(plat_data->dev);
-
+#endif
 	return 0;
 
 comp_err:
 	snd_soc_unregister_component(&pdev->dev);
 	snd_sof_free_topology(sdev);
 fw_run_err:
 	snd_sof_fw_unload(sdev);
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_VIRTIO_FE)
```

Comment:
Can we remove a lot of the conditional compilation and use runtime flags in sdev for is_vfe and is_vbe ?
I know we cant do this everywhere atm due to buggy headers, but we should do this where we can so that less code needs updated once the ACRN headers are fixed.